### PR TITLE
[RFC][UN-11734] Add new uni datepicker range

### PR DESCRIPTION
**Pull request by asantos00**
# Issue
https://uniplaces.atlassian.net/browse/UN-11734

# What this does?
This ports part of the calendar done beautifully from @AndreJoaquim in the offer page to ember cli uniq. It is a range calendar and does some of the basic functionality.

Most of the code is his, 🙏 

# Screenshots
screen shot 2018-01-25 at 01 58 04
screen shot 2018-01-25 at 01 57 57

### CODE REVIEW

File: addon/styles/partials/_uni-datepicker-range.scss [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+            &--range-end,
+            &--last-highlighted {
+                background-image: url(assetsRootURL() + "assets/images/ember-cli-uniq/last-highlighted.svg");
+                background-position: 2px 6px;
+            }
+        }
+
+        &-nav {
+            @include flex();
```
Comment (mmelo) [inline addon/styles/partials/_uni-datepicker-range.scss]: change these includes as we no longer use'em

File: addon/styles/partials/_uni-datepicker-range.scss [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                text-transform: uppercase;
+                color: #8ca6b2;
+                user-select: none;
+            }
+        }
+    }
+
+    .calendar__nav {
+        @include flex();
+        @include align-items-start();
```
Comment (mmelo) [inline addon/styles/partials/_uni-datepicker-range.scss]: same here mate

File: app/components/uni-datepicker-range-day.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from 'ember-cli-uniq/components/uni-datepicker-range-day';
```
Comment (mmelo) [inline app/components/uni-datepicker-range-day.js]: newline @EOF

File: app/components/uni-datepicker-range.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export { default } from 'ember-cli-uniq/components/uni-datepicker-range';
```
Comment (mmelo) [inline app/components/uni-datepicker-range.js]: same here sir

File: public/assets/images/icons/calendar-arrow.svg [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 10 16"><path fill="#224251" fill-rule="evenodd" d="M1.278 15.586L0 14.046l7.446-6.253L0 1.54 1.278 0l8.364 7.023a1.007 1.007 0 0 1 0 1.54l-8.364 7.023z"/></svg>
+
```
Comment (mmelo) [inline public/assets/images/icons/calendar-arrow.svg]: remove empty line please

File: public/assets/images/icons/first-highlighted.svg [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="27" height="24" viewBox="0 0 27 24">
+    <path fill="#00ADEF" fill-rule="evenodd" d="M23 8V3a3 3 0 0 0-3-3H3a3 3 0 0 0-3 3v18a3 3 0 0 0 3 3h17a3 3 0 0 0 3-3v-5l3.293-3.293a1 1 0 0 0 0-1.414L23 8z"/>
+</svg>
+
```
Comment (mmelo) [inline public/assets/images/icons/first-highlighted.svg]: same here

File: public/assets/images/icons/last-highlighted.svg [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="27" height="24" viewBox="0 0 27 24">
+    <path fill="#00ADEF" fill-rule="evenodd" d="M3.586 16v5a3 3 0 0 0 3 3h17a3 3 0 0 0 3-3V3a3 3 0 0 0-3-3h-17a3 3 0 0 0-3 3v5L.293 11.293a1 1 0 0 0 0 1.414L3.586 16z"/>
+</svg>
+
```
Comment (mmelo) [inline public/assets/images/icons/last-highlighted.svg]: and here

File: public/assets/images/uniplaces_logo.svg [hunk 8]
```diff
@@ -1,62 +1 @@
...
...60.96-68.086-60.96-76.782 0-111.619 54.24-118.745 110.049-6.333 53.834 17.021 99.36 87.471 99.36 33.743 0 61.031-6.456 73.272-10.266 2.822-.882 4.692-3.528 4.674-6.491l-.282-32.491c-.035-4.48-4.322-7.691-8.625-6.403-12.489 3.759-28.54 6.563-44.098 6.563M1638.961 219.88c5.415-11.606 11.43-21.096 18.38-29.492 3.069-3.704 2.805-9.26-2.17-11.536-12.647-4.974-36.742-10.636-58.155-10.636-55.404 0-104.211 27.905-113.506 104.599-8.908 73.995 33.32 104.211 86.007 104.211 16.246 0 30.78-1.729 46.32-5.539.017-.017.035-.017.053-.017.282-.071.565-.159.829-.212 8.273-2.399 15.275-8.96 15.522-18.627l.582-22.49c.106-4.427-4.057-7.743-8.379-6.685-15.857 3.846-28.152 5.539-39.052 5.539-26.723 0-43.004-11.236-37.571-55.792 5.415-43.003 23.248-54.998 50.359-54.998 8.537 0 24.783 3.51 33.179 5.468 3.121.741 6.244-.882 7.602-3.793M1263.213 327.054c-11.236 0-19.368-5.027-17.427-20.919l24.236-196.974c.511-4.022-2.628-7.567-6.685-7.567h-52.264c-3.404 0-6.279 2.54-6.685 5.909l-24.853 202.125c-5.027 40.675 5.045 67.398 49.971 67.398 6.262 0 25.241-1.835 45.156-9.631 3.775-1.482 5.327-5.927 3.475-9.525-4.233-8.29-7.373-17.515-9.419-27.658-.388-1.923-2.064-3.351-4.004-3.21-.513.035-1.025.052-1.501.052"/></svg>
```
Comment (mmelo) [inline public/assets/images/uniplaces_logo.svg]: newline @EOF

File: public/assets/images/uniplaces_logo_white.svg [hunk 9]
```diff
@@ -1,62 +1 @@
...
...6-68.086-60.96-76.782 0-111.619 54.24-118.745 110.049-6.333 53.834 17.021 99.36 87.471 99.36 33.743 0 61.031-6.456 73.272-10.266 2.822-.882 4.692-3.528 4.674-6.491l-.282-32.491c-.035-4.48-4.322-7.691-8.625-6.403-12.489 3.759-28.54 6.563-44.098 6.563M1638.961 219.88c5.415-11.606 11.43-21.096 18.38-29.492 3.069-3.704 2.805-9.26-2.17-11.536-12.647-4.974-36.742-10.636-58.155-10.636-55.404 0-104.211 27.905-113.506 104.599-8.908 73.995 33.32 104.211 86.007 104.211 16.246 0 30.78-1.729 46.32-5.539.017-.017.035-.017.053-.017.282-.071.565-.159.829-.212 8.273-2.399 15.275-8.96 15.522-18.627l.582-22.49c.106-4.427-4.057-7.743-8.379-6.685-15.857 3.846-28.152 5.539-39.052 5.539-26.723 0-43.004-11.236-37.571-55.792 5.415-43.003 23.248-54.998 50.359-54.998 8.537 0 24.783 3.51 33.179 5.468 3.121.741 6.244-.882 7.602-3.793M1263.213 327.054c-11.236 0-19.368-5.027-17.427-20.919l24.236-196.974c.511-4.022-2.628-7.567-6.685-7.567h-52.264c-3.404 0-6.279 2.54-6.685 5.909l-24.853 202.125c-5.027 40.675 5.045 67.398 49.971 67.398 6.262 0 25.241-1.835 45.156-9.631 3.775-1.482 5.327-5.927 3.475-9.525-4.233-8.29-7.373-17.515-9.419-27.658-.388-1.923-2.064-3.351-4.004-3.21-.513.035-1.025.052-1.501.052"/></g></svg>
```
Comment (mmelo) [inline public/assets/images/uniplaces_logo_white.svg]: same here mate

**Review by mmelo — CHANGES_REQUESTED** [review]
few minor changes, overall great work 👌

**Review by telmaantunes — COMMENTED** [review]
So, I don't know if you are aware but the current calendar in offer page is going through a lot of changes, one of them is the highlighted, that does not work like before. 
Can you please check these layouts and the comments and talk with Fabiano and Cardoso to see if it makes sense to have different calendars? https://app.zeplin.io/project/590c4b4569af7c0dcc4a2575/screen/5a2e8bc0e255a010384a1653?
https://app.zeplin.io/project/590c4b4569af7c0dcc4a2575/screen/5a2e8bceb67ba51760631543?

**Comment by asantos00** [comment]
@telmaantunes super pertinent, I knew about that through @mumia. However, this just exports a much simpler calendar to the addon, even though I think the styles should match (the handle and the selected state and blablabla). Ideally, one day (**I dream** 😂) we would be able to have the simpler calendar being used inside the much complex one and they would share styles and so.

At the moment I think it is too much to *loose* more time with this (at least in spa-search), however, I will talk to Fabiano to understand about the next iterations of this calendar (that will almost for sure match the offer page one)

**Review by AndreJoaquim — APPROVED** [review]

**Review by mmelo — APPROVED** [review]

**Review by mumia — APPROVED** [review]

**Pull request merged**